common/persistence/dataInterfaces.go

```diff
@@ -1083,10 +1060,6 @@ type (
 		AppendHistoryNodes(ctx context.Context, request *AppendHistoryNodesRequest) (*AppendHistoryNodesResponse, error)
 		// AppendRawHistoryNodes add a node of raw histories to history node table
 		AppendRawHistoryNodes(ctx context.Context, request *AppendRawHistoryNodesRequest) (*AppendHistoryNodesResponse, error)
-		// ParseHistoryBranchInfo parses the history branch for branch information
-		ParseHistoryBranchInfo(ctx context.Context, request *ParseHistoryBranchInfoRequest) (*ParseHistoryBranchInfoResponse, error)
-		// UpdateHistoryBranchInfo updates the history branch with branch information
-		UpdateHistoryBranchInfo(ctx context.Context, request *UpdateHistoryBranchInfoRequest) (*UpdateHistoryBranchInfoResponse, error)
 		// NewHistoryBranch initializes a new history branch
 		NewHistoryBranch(ctx context.Context, request *NewHistoryBranchRequest) (*NewHistoryBranchResponse, error)
```

Comment:
This method should be removed as well? Same of the corresponding metric def and impls?

Reply:
I think this method also belongs to the new HistoryBranchUtil

Reply:
The `ParseHistoryBranchInfo()` and `UpdateHistoryBranchInfo()` helpers operate on the same function signature and are store agnostic. However, `NewHistoryBranch()` is tied to the specific store implementation. While it is technically possible to move this from the interface into `HistoryBranchUtil`, it actually makes the code layout for specific store implementations more hairy.